os/fs/smartfs/README.txt

```diff
@@ -141,59 +239,92 @@ General operation
+  operations that may have started (Checked-in) but were not completed
+  (Checked-out) before power was lost.
+
+  The structure of a journal entry is described as below:
+  ----------------------------------------------------------------------------
+  |	  Sector Header        |	      Journal Metadata		     |
+  ----------------------------------------------------------------------------
+  |<6 bytes as described above>| CRC-16(2) | PSector(2) | Seq(2) | Status(1) |
+  ----------------------------------------------------------------------------
+
+    - CRC-16 - error detecting checksum code
+    - PSector - physical location where change is made
+    - Seq - Journal entry's index
+    - Status
+        -- First 4 bits indicate journal state (Checked-in or Checked-out)
+        -- Last 4 bits indicate entry purpose / kind of operation being logged
+           (commit / release / erase)
+
   Wear Leveling
   =============
 
   When wear leveling is enabled, the code automatically writes data across
   the entire FLASH device in a manner that causes each erase block to be
-  worn (i.e. erased) evenly.  This is accomplished by maintaining a 4-bit
+  worn (i.e. erased) evenly. This is accomplished by maintaining a 4-bit
   wear level count for each erase block and forcing less worn blocks to be
-  used for writing new data.  The code maintains each block's erase count
-  to be within 16 erases of each other, though through testing, the span
-  so far was never greater than 10 erases of each other.
+  used for writing new data. The code maintains each block's erase count
+  to be within 16 erases of each other and through thorough testing, it is
+  observed that the difference between erase counts of any 2 sectors is not
+  more than 10%.
 
   As the data in a block is modified repeatedly, the erase count will
   increase.  When the wear level reaches a value of 8 or higher, and the block
   needs to be erased (because the data in it has been modified, etc.) the code
   will select an erase block with the lowest wear count and relocate it to
-  this block (with the higher wear count).  The idea being that a block with
+  this block (with the higher wear count). The idea being that a block with
   the lowest wear count contains more "static" data and should require fewer
-  additional erase operations.  This relocation process will continue on the
+  additional erase operations. This relocation process will continue on the
   block (only when it needs to be erased again).
 
+  It is noteworthy that SMARTFS employs Static wear levelling, i.e. it is
+  capable of erasing an eraseblock with active file data by relocating
+  the sector containing that data to another eraseblock. This allows it to
+  move the data belonging to files that may be stored in the filesystem for
+  a very long period of time and thereby erasing those sectors as well, which
+  ultimately helps in providing better wear dsitribution.
```

Comment:
Please fix the typo
`dsitribution` -> `distribution`

Reply:
Will do, thankyou.